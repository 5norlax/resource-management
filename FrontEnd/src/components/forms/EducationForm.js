import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';


const EducationForm = ({ index, removeEducation, School, Location, Degree, Major, Minor, GradDate }) => {
    var today = new Date();
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="school">School</Label>
                    <Input type="text" name="school" id="school" placeholder="Education/University" value={School} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="location">Location</Label>
                    <Input type="text" name="location" id="location" placeholder="University Location" value={Location} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Degree</Label>
                    <Input type="select" name="degree" id="degree" placeholder="Degree Obtained" value={Degree}>
                        <option value='' selected disabled>Please Select</option>
                        <option>Associate</option>
                        <option>Bacholor</option>
                        <option>Master</option>
                        <option>PhD</option>
                    </Input>
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Major</Label>
                    <Input type="text" name="degree" id="degree" placeholder="Declared Major" value={Major} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="gradDate">Graduation Date</Label>
                    <Input type="number" min="1900" max={today.getFullYear()} step="1" placeholder="Graduation Year" value={GradDate} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => {removeEducation(index)}} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default EducationForm