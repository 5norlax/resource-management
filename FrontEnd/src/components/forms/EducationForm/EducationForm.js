import React from 'react'
import { Label, Input, FormGroup, Row, Button } from 'reactstrap'

export default React.memo(function EducationForm({
  index,
  removeEducation,
  School,
  Location,
  Degree,
  Major,
  Minor,
  GradDate,
  handleEducationChange,
}) {
  return (
    <div>
      <Row>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          <Label for="School">School</Label>
          <Input
            required
            type="text"
            name="School"
            id="School"
            placeholder="Education/University"
            value={School}
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          />
        </FormGroup>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          <Label for="Location">Location</Label>
          <Input
            required
            type="text"
            name="Location"
            id="Location"
            placeholder="University Location"
            value={Location}
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          />
        </FormGroup>
        <FormGroup className="col-xl-3 col-lg-3 col-md-6 col-sm-12 col-xs-12">
          <Label for="Degree">Degree</Label>
          <Input
            required
            type="select"
            name="Degree"
            id="Degree"
            placeholder="Degree Obtained"
            value={Degree}
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          >
            <option value="" defaultValue disabled>
              Please Select
            </option>
            <option>Associate</option>
            <option>Bachelor</option>
            <option>Master</option>
            <option>PhD</option>
          </Input>
        </FormGroup>
        <FormGroup className="col-xl-3 col-lg-3 col-md-6 col-sm-12 col-xs-12">
          <Label for="GradDate">Graduation Date</Label>
          <Input
            required
            type="date"
            name="GradDate"
            id="GradDate"
            value={GradDate}
            data-testid="educationDateInput"
            placeholder="date placeholder"
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          />
        </FormGroup>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          <Label for="Major">Major(s)</Label>
          <Input
            required
            type="text"
            name="Major"
            id="Major"
            placeholder="Declared Major(s)"
            value={Major}
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          />
        </FormGroup>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          <Label for="Minor">Minor(s)</Label>
          <Input
            type="text"
            name="Minor"
            id="Minor"
            placeholder="Declared Minor(s)"
            value={Minor}
            onChange={(e) => {
              handleEducationChange(index, e)
            }}
          />
        </FormGroup>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
          <Button
            onClick={() => {
              removeEducation(index)
            }}
            style={{ marginTop: '32px' }}
            color="danger"
          >
            Remove Education
          </Button>
        </FormGroup>
      </Row>
      <hr />
    </div>
  )
})
