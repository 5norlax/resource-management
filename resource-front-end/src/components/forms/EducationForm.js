import React from 'react';
import { Label, Input, FormGroup, Row } from 'reactstrap';


const EducationForm = () => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="school">School</Label>
                    <Input type="text" name="school" id="school" placeholder="Education/University" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="location">Location</Label>
                    <Input type="text" name="location" id="location" placeholder="University Location" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Degree</Label>
                    <Input type="select" name="degree" id="degree" placeholder="Degree Obtained">
                        <option>Associate</option>
                        <option>Bacholor</option>
                        <option>Master</option>
                        <option>PhD</option>
                    </Input>
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Major</Label>
                    <Input type="text" name="degree" id="degree" placeholder="Declared Major" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="gradDate">Graduation Date</Label>
                    <Input type="text" name="gradDate" id="gradDate" placeholder="Graduation Date" />
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default EducationForm