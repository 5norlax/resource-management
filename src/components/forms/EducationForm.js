import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';


const EducationForm = ({index, removeEducation}) => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="school">School</Label>
                    <Input type="text" name="school" id="school" placeholder="Education/University" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="location">Location</Label>
                    <Input type="text" name="location" id="location" placeholder="University Location" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Degree</Label>
                    <Input type="select" name="degree" id="degree" placeholder="Degree Obtained">
                        <option value='' selected disabled>Please Select</option>
                        <option>Associate</option>
                        <option>Bacholor</option>
                        <option>Master</option>
                        <option>PhD</option>
                    </Input>
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="degree">Major</Label>
                    <Input type="text" name="degree" id="degree" placeholder="Declared Major" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="gradDate">Graduation Date</Label>
                    <Input
                    type="date"
                    name="gradDate"
                    id="gradDate"
                    placeholder="date placeholder"
                    />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => {removeEducation(index)}} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default EducationForm