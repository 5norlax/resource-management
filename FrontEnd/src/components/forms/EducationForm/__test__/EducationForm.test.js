import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import EducationForm from '../EducationForm';

afterEach(cleanup);

it('matches snapshot', async () => {
  const tree = renderer.create(<EducationForm />).toJSON()
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<EducationForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Input Fields', () => {
  const handleEducationChange = jest.fn();
  it('School Input triggers handleEducationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const schoolInput = queryByPlaceholderText('Education/University');
    fireEvent.change(schoolInput, {target: {value: "test"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
  it('Location Input triggers handleEducationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const locationInput = queryByPlaceholderText('University Location');
    fireEvent.change(locationInput, {target: {value: "test"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
  it('Degree Input triggers handleEducationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const degreeInput = queryByPlaceholderText('Degree Obtained');
    fireEvent.change(degreeInput, {target: {value: "Bachelor"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
  it('Date Input triggers handleEducationChange onChange', async () => {
    const {queryByTestId} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const dateInput = queryByTestId('educationDateInput');
    fireEvent.change(dateInput, {target: {value: "0001-01-01"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
  it('Major(s) Input triggers handleEducationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const majorInput = queryByPlaceholderText('Declared Major(s)');
    fireEvent.change(majorInput, {target: {value: "test"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
  it('Minor(s) Input triggers handleEducationChange onChange', async () => {
    const {queryByPlaceholderText} = render(<EducationForm handleEducationChange={handleEducationChange} />);
    const minorInput = queryByPlaceholderText('Declared Minor(s)');
    fireEvent.change(minorInput, {target: {value: "test"}});
    expect(handleEducationChange).toHaveBeenCalled();
  });
});

describe('Remove Button', () => {
  it('Button Removes Education onClick', async () => {
    const removeEducation = jest.fn();
    const {queryByText} = render(<EducationForm removeEducation={removeEducation} />);
    fireEvent.click(queryByText('Remove Education'));
    expect(removeEducation).toHaveBeenCalled();
  });
});