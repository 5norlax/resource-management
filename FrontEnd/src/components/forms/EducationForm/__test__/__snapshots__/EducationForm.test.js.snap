// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="School"
      >
        School
      </label>
      <input
        className="form-control"
        id="School"
        name="School"
        onChange={[Function]}
        placeholder="Education/University"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Location"
      >
        Location
      </label>
      <input
        className="form-control"
        id="Location"
        name="Location"
        onChange={[Function]}
        placeholder="University Location"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-3 col-lg-3 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Degree"
      >
        Degree
      </label>
      <select
        className="form-control"
        id="Degree"
        name="Degree"
        onChange={[Function]}
        placeholder="Degree Obtained"
        required={true}
      >
        <option
          defaultValue={true}
          disabled={true}
          value=""
        >
          Please Select
        </option>
        <option>
          Associate
        </option>
        <option>
          Bachelor
        </option>
        <option>
          Master
        </option>
        <option>
          PhD
        </option>
      </select>
    </div>
    <div
      className="col-xl-3 col-lg-3 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="GradDate"
      >
        Graduation Date
      </label>
      <input
        className="form-control"
        data-testid="educationDateInput"
        id="GradDate"
        name="GradDate"
        onChange={[Function]}
        placeholder="date placeholder"
        required={true}
        type="date"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Major"
      >
        Major(s)
      </label>
      <input
        className="form-control"
        id="Major"
        name="Major"
        onChange={[Function]}
        placeholder="Declared Major(s)"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="Minor"
      >
        Minor(s)
      </label>
      <input
        className="form-control"
        id="Minor"
        name="Minor"
        onChange={[Function]}
        placeholder="Declared Minor(s)"
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 text-right form-group"
    >
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "marginTop": "32px",
          }
        }
        type="button"
      >
        Remove Education
      </button>
    </div>
  </div>
  <hr />
</div>
`;
